tests/eko/test_ad_lo.py

```diff
@@ -12,39 +13,63 @@ def test_number_conservation():
     # number
     N = complex(1.0, 0.0)
     s1 = harmonics.harmonic_S1(N)
-    np.testing.assert_almost_equal(ad_lo.gamma_ns_0(N, s1), 0)
+    np.testing.assert_almost_equal(ad_as1.gamma_ns(N, s1), 0)
 
 
 def test_quark_momentum_conservation():
     # quark momentum
     N = complex(2.0, 0.0)
     s1 = harmonics.harmonic_S1(N)
-    np.testing.assert_almost_equal(ad_lo.gamma_ns_0(N, s1) + ad_lo.gamma_gq_0(N), 0)
+    np.testing.assert_almost_equal(
+        ad_as1.gamma_ns(N, s1) + ad_as1.gamma_gq(N),
+        0,
+    )
+    np.testing.assert_almost_equal(
+        ad_aem1.gamma_ns(N, s1) + ad_aem1.gamma_phq(N),
+        0,
+    )
 
 
 def test_gluon_momentum_conservation():
     # gluon momentum
     N = complex(2.0, 0.0)
     s1 = harmonics.harmonic_S1(N)
     np.testing.assert_almost_equal(
-        ad_lo.gamma_qg_0(N, NF) + ad_lo.gamma_gg_0(N, s1, NF), 0
+        ad_as1.gamma_qg(N, NF) + ad_as1.gamma_gg(N, s1, NF), 0
     )
 
 
 def test_gamma_qg_0():
     N = complex(1.0, 0.0)
     res = complex(-20.0 / 3.0, 0.0)
-    np.testing.assert_almost_equal(ad_lo.gamma_qg_0(N, NF), res)
+    np.testing.assert_almost_equal(ad_as1.gamma_qg(N, NF), res)
 
 
 def test_gamma_gq_0():
     N = complex(0.0, 1.0)
     res = complex(4.0, -4.0) / 3.0
-    np.testing.assert_almost_equal(ad_lo.gamma_gq_0(N), res)
+    np.testing.assert_almost_equal(ad_as1.gamma_gq(N), res)
 
 
 def test_gamma_gg_0():
     N = complex(0.0, 1.0)
     s1 = harmonics.harmonic_S1(N)
     res = complex(5.195725159621, 10.52008856962)
-    np.testing.assert_almost_equal(ad_lo.gamma_gg_0(N, s1, NF), res)
+    np.testing.assert_almost_equal(ad_as1.gamma_gg(N, s1, NF), res)
+
+
+def test_gamma_phq_0():
+    N = complex(0.0, 1.0)
+    res = complex(4.0, -4.0) / 3.0 / 4 * 3
+    np.testing.assert_almost_equal(ad_aem1.gamma_phq(N), res)
+
+
+def test_gamma_qph_0():
+    N = complex(1.0, 0.0)
+    res = complex(-20.0 / 3.0, 0.0) * 3 / 0.5
+    np.testing.assert_almost_equal(ad_aem1.gamma_qph(N, NF), res)
+
+
+def test_gamma_phph_0():
+    res = complex(-4.0 / 3, 0.0)
+    np.testing.assert_almost_equal(ad_aem1.gamma_phph(), res)
```

Comment:
Here we can't test the photon conservation?  
I see that `gamma_qph` contains the factor NC, I was wondering if this has to be included also in `gamma_phph`
according to eq 2.4 of `Carrazza:2015dea`

Reply:
I agree with you: 
in fact we had included NC in the def of gamma_qph so that we can remove it from the def of theta+- (see eq. 2.27 of Carrazza:2015dea). But the same applies gamma_phph with e_\Sigma.
In my opinion either we put both NC factors inside gamma_qph and gamma_phph or we put them in theta+- and e_\Sigma.

Reply:
If think that `gamma_phph` should be:
```
- 2 / 3 * NC * 2 * nf
```

this way the photon momentum conservation test will work

Reply:
@niclaurenti. Just for the records, don't know if this is already in the references: 
[paper](https://arxiv.org/pdf/1512.00612.pdf), but might come useful also in the future.

Reply:
Thanks, @giacomomagni, this appeared at the same time as Stefano's thesis, so they might have a common reference, but it's spelling out the full "algorithm to obtain QED ad from QCD ones", that might become handy sooner or later :)